bpf: Do not delete and re-add ip rules and routes in the proxy table

tgraf opened the pull request:
This can otherwise lead to a brief interruption of traffic from the proxy to
local endpoints when the agent is being restarted.

Related: #5651

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/5653)
<!-- Reviewable:end -->

joestringer commented on bpf/init.sh:
```diff
@@ -140,19 +125,15 @@ function move_local_rules()
-		# Any packet from a proxy uses a separate routing table
-		ip -4 rule add fwmark 0xA00/0xF00 pref 10 lookup $PROXY_RT_TABLE
+	if [ -z "$(ip -4 rule list $rulespec)" ]; then
+		ip -4 rule add $rulespec
```
Do these need to be `|| true` to handle `EEXIST`?

```
$ sudo ip rule add fwmark 0xA00/0xF00 pref 10 lookup 2005
$ echo $?
0
$ sudo ip rule add fwmark 0xA00/0xF00 pref 10 lookup 2005
RTNETLINK answers: File exists
$ echo $?
2
```

tgraf replied:
That's why it checks if the rule already exists with `ip rule list`. Unfortunately there is no `ip rule replace`.

joestringer reviewed (changes requested):
Seems like these scripts will exit early / fail out if the ip rules already exist.

tgraf committed:
bpf: Do not delete and re-add ip rules and routes in the proxy table

This can otherwise lead to a brief interruption of traffic from the proxy to
local endpoints when the agent is being restarted.

Signed-off-by: Thomas Graf <thomas@cilium.io>
